crml/staking/src/lib.rs

```diff
@@ -1399,25 +1401,17 @@ impl<T: Trait> Module<T> {
 		let now = T::Time::now();
 		let previous_era_start = <CurrentEraStart<T>>::mutate(|v| sp_std::mem::replace(v, now));
 		let era_duration = now - previous_era_start;
+		<CurrentEraDuration<T>>::put(era_duration);
```

Comment:
Does this need to be stored? can we pass it as a parameter into `current_total_payout` instead

Reply:
Actually the `era_duration` is the key parameter we want to keep it same within the `new_era` function,  if we pass from outside, the test value can not be same with the one in `new_era`

Reply:
When reusing `fn current_total_payout` for tests, it's not clear what era_duration we need to pass

Reply:
It's possible to control the the duration in tests by configuring the timestamps `T::Time::now()` and `CurrentEraStart` values. This would allow tests to use some fixed duration for eras

Reply:
I believe the tests are configuring the timestamps of `T::Time::now()` by using `Timestamp::set_timestamp((System::block_number() * 1000).into());`, is that correct?

I suggest renaming from `CurrentEraDuration` to `PreviousEraDuration` - the value we are storing is the delta t between the start of the new era and the start of the previous era.

Reply:
> It's possible to control the the duration in tests by configuring the timestamps `T::Time::now()` and `CurrentEraStart` values. This would allow tests to use some fixed duration for eras

For the mock config in `multi_token_economy_tests`, it does config fix duration, but for our test we use `runtime` directly, so we don't find a good way to solve this

Reply:
Will be addressed in a follow up PR (if possible)